### PR TITLE
Bcode compare

### PR DESCRIPTION
**Pull request by romanowski**
Add option `--baseline` that when provided compiles corpus with provided scalac. It will also compare binary files against 2 compilation (scala nad baseline)

Also:
- remove nested subrepo (it was pain even to remove it)
- updates scala-library to version from PR is perf-tester

### CODE REVIEW

**Comment by romanowski** [comment]
Bytecode  generated always differ on `scala.throws` annotation. Supprisingly adding umad to compiler causing difference (if both scala and baseline are run with or without there is no difference). Investigation showed that when running with umad picke for object scala.throws is missing:

```
80 = {Types$UniqueTypeBounds} " <: Throwable"
81 = {Types$AliasNoArgsTypeRef} "Throwable"
82 = {Symbols$AliasTypeSymbol} "type Throwable"
```

File: run.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 parser.add_option('-b', '--baseline-scala', dest='baseline', default="", help='Path to the baseline Scala build dir')
 parser.add_option('-d', '--debugPort', dest='debugPort', help='Port on which remote debugger can be attached')
 parser.add_option('-c', '--corpus', dest='corpus', default="scala-library", help='Project to compile')
+parser.add_option('-o', '--overrides', dest='overrides', default="", help='Config overrides (key=values paris separated by ;')
```
Comment (pkukielka) [inline run.py]: Please use more meaningful long name.
Also it's quite generic, can override any jvm option, so is there sense to describe it as config change?
Also: https://stackoverflow.com/questions/7671383/creating-an-array-from-a-command-line-option-pythonoptparse?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

File: noop/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# noop - empty compiler plugin
```
Comment (pkukielka) [inline noop/README.md]: Cannot we just call umad but exclude everything?

File: run.py [hunk 3]
```diff
@@ -38,18 +40,49 @@ def findFiles(path, regex):
 
 debugOptions = []
 if options.debugPort:
-    debugOptions =  ["-J-agentlib:jdwp=transport=dt_socket,server=n,address=localhost:{},suspend=y".format(options.debugPort)]
```
Comment (pkukielka) [inline run.py:41]: Incorrect change. Possibly you can add default to `--debugPort` though.

File: run.py [hunk 4]
```diff
@@ -38,18 +40,49 @@ def findFiles(path, regex):
...
+outputBase = tempfile.mkdtemp()
+scalaOutput = os.path.join(outputBase, "scala")
+baselineOutput = os.path.join(outputBase, "baseline")
+
+os.mkdir(scalaOutput)
+os.mkdir(baselineOutput)
+
+def call_compiler(scalaLocation, output, additionalOptions = []):
+	subprocess.call([
+						os.path.join(scalaLocation, "bin", "scalac"),
```
Comment (pkukielka) [inline run.py]: What happened with that formatting? :P

File: umad/src/main/java/com/pkukielka/AccessMonitorTransformer.java [hunk 5]
```diff
@@ -10,6 +10,8 @@
 public class AccessMonitorTransformer implements ClassFileTransformer {
     private final ClassMethodSelector classMethodSelector;
     private final MethodRewriter methodRewriter;
+    private static final AppConfig conf = new AppConfig();
+
```
Comment (pkukielka) [inline umad/src/main/java/com/pkukielka/AccessMonitorTransformer.java]: Remove if you will be somewhere around this file ;)

File: run.py (hunk 3 above)
Comment (romanowski) [reply run.py:41]: reverted

File: noop/README.md (hunk 2 above)
Comment (romanowski) [reply noop/README.md]: Done

File: umad/src/main/java/com/pkukielka/AccessMonitorTransformer.java (hunk 5 above)
Comment (romanowski) [reply umad/src/main/java/com/pkukielka/AccessMonitorTransformer.java]: it is changed now

**Pull request merged**